File: ts/management-api-v2/cm_api_v2_patch_asset_folders.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client.modifyAssetFolders()
  .withData(
    [
      {
        op: 'addinto',
        value: {
          external_id: 'folder-with-shared-assets',
          codename: 'shared_assets',
          name: 'Shared assets',
          folders: []
        },
        before: {
          external_id: 'folder-with-downloadable-assets'
        }
      },
      {
        op: 'remove',
        reference: {
          external_id: 'folder-with-downloadable-archives'
        }
      },
      {
        op: 'rename',
        value: 'Legal documents',
        reference: {
          codename: 'folder_documents'
        }
      }
    ]
  )
  .toPromise();

File: net/management-api-v2/cm_api_v2_patch_asset_folders.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.AssetFolders;
using Kontent.Ai.Management.Models.AssetFolders.Patch;
using Kontent.Ai.Management.Models.Shared;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = (await client.ModifyAssetFoldersAsync(
[
    new AssetFolderAddIntoPatchModel
    {
        Reference = Reference.ByExternalId("folder-with-shared-asset"),
        Value = new AssetFolderHierarchy
        {
            ExternalId = "folder-with-shared-assets",
            Name = "Shared assets",
            Codename = "shared_assets",
        },
        Before = Reference.ByExternalId("folder-with-downloadable-assets")
    },
    new AssetFolderRemovePatchModel
    {
        Reference = Reference.ByExternalId("folder-with-archived-assets")
    },
    new AssetFolderRenamePatchModel
    {
        Reference = Reference.ByCodename("folder_documents"),
        Value = "Legal documents"
    }
])).EnsureSuccess();
